libs/sdk-core/src/persist/transactions.rs

```diff
@@ -101,6 +104,38 @@ impl SqliteStorage {
         Ok(())
     }
 
+    /// Selectively updates the external metadata fields of a payment
+    pub fn set_payment_external_metadata(
+        &self,
+        payment_hash: String,
+        new_metadata: String,
+    ) -> PersistResult<()> {
+        if new_metadata.len() > METADATA_MAX_SIZE_KB {
+            return Err(PersistError::Generic(
+                anyhow::anyhow!("Max metadata size ({}MB) has been exceeded", METADATA_MAX_SIZE_KB / 1000)
+            ));
+        }         
```

Comment:
Could replace this with the ensure_sdk! macro
```suggestion
        ensure_sdk!(
            new_metadata.len() <= METADATA_MAX_SIZE_KB,
            PersistError::Generic(
                anyhow!("Max metadata size ({}MB) has been exceeded", METADATA_MAX_SIZE_KB / 1000)
            )
        );         
```

Reply:
Didn't know about it. Thanks Ross!